capsules/src/dals/README.md

```diff
@@ -0,0 +1,100 @@
+segment-by-segment through **DALS** which places it in an empty space in flash space and
+then sets up the process-specific elements for the newly-loaded app.
+
+## Design
+
+The design of **DALS** consists of a series of capsules with one capsule, the **AppLoader**, considered
+to be trusted.  This main trusted capsule is the heart of **DALS** and contains the code which collects
+the raw app data from another capsule, moves it through a pipeline of steps, places the
+resulting app binary in an empty slot in flash space, and, finally, sets up its process-memory region
+so the scheduler can run it at the appropriate time.
+
+Basic Pipeline at the core of **DALS**.
+
+
+Collect raw data from peripheral capsule (BLE, Zigbee, etc.) -----> Decompress Data  ----->  Find empty slot in flash --- 
+
+---> Verify Data ------> Validate Data ----> Set up process-specific structures (stack, heap, etc.)
+
+
+The trusted `AppLoader` capsule interfaces with 4 differennt peripheral capsules. These
+peripheral capsules consist of the **AppLoaderClient**, **Decompressor**, **Verifier**, and 
+**Validator** capsules.  Each capsule is named after the trait or interface it should implement.
+These interfaces are defined in `interfaces.rs`.
+
+The way these capsules are interconnected is diagrammed below.
+
+
+                                            AppLoaderClient
+                                                 |
+                                                 |
+                                                 |
+                                                 |
+                         Validator ---------  AppLoader  ----------  Decompressor
+                                                 |
+                                                 |
+                                                 |
+                                                 |
+                                              Verifier
+                                        
+
+There exists a circular reference between the **AppLoader** capsule and each of the peripheral
+capsules.  This enables the **AppLoader** to call functions implemented by a peripheral capsule and 
+vice versa.
+
+The peripheral modules are detailed below.  Please see `interfaces.rs` for more information on functions and such.
+
+
+## AppLoaderClient
+
+This capsule implements `pub trait AppLoaderClient<'a>`.  This capsule can be any capsule
+which receives data via standard communication protocols for embedded systems such as BLE, 
+Zigbee, USB.  This capsule's main function is to send this raw received data buffer-by-buffer
+to the **AppLoader** capsule.  In order to prevent bottleneck issues, this capsule must send an ack 
+only after its buffer is returned to it from the **AppLoader**.
+
+This capsule must also include a buffer the size of one data segment (different for different protocols) which
+it sends to the **AppLoader** to send one segment of the received data.
+
+## Decompressor
```

Comment:
Does the decompressor write the data to flash?

Reply:
No. the decompressor will put the decompressed data into a buffer which AppLoader will read from.  Ill make it clearer in the README.